### ISSUE TITLE
How to parse output the OUTCAR file generated by ML method

### ISSUE THREAD

**Issue by Guodonglin-cqust** [flokno/tools.tdep#11]
Dear developer

     For normal OUTCAR file, i can parse output the OUTCAR using the tdep_parse_output.py, but for OUTCAR generated by ML method, i can not parse output the OUTCAR file, could you help me. Thank you very much.

    The attach file is the  OUTCAR generated by ML method.

Besh wishes

Donglin

**Comment by Guodonglin-cqust** [comment]
[OUTCAR0001.zip](https://github.com/user-attachments/files/15525358/OUTCAR0001.zip)

**Comment by flokno** [comment]
Hi @Guodonglin-cqust ,
We generally recommend the `vasprun.xml` for parsing — do you have this file?

**Comment by Guodonglin-cqust** [comment]
I have the vasprun.xml, and the attach file is the  vasprun.xml generated by ML method.
[vasprun.zip](https://github.com/user-attachments/files/15525473/vasprun.zip)

**Comment by Guodonglin-cqust** [comment]
Based on the vasprun.xml generated by ML method, i also can not parse output the vasprun.xml file

(base) cqwl111@cqwl111:~/Videos/456$ tdep_parse_output vasprun.xml
Parse 1 file(s)
... empty forces will be ignored: False
*** SIMULATION TEMPERATURE IS NOT GIVEN
--> set to -314.15K to remind you
... parse file   1: 'vasprun.xml'
... found 0 samples
... discard 0 steps at beginning
... no data found, abort.
(base) cqwl111@cqwl111:~/Videos/456$

**Comment by flokno** [comment]
Hi @Guodonglin-cqust , there seems to be an issue with parsing this output. Since we use ASE for parsing, I would transfer this to the ASE issue tracker: https://gitlab.com/ase/ase/-/issues . Is it ok if I share your output there?

**Comment by Guodonglin-cqust** [comment]
OK， thank you very much.

**Comment by Guodonglin-cqust** [comment]
Could you help me revise the tdep_parse_output.py, making it to parse output the OUTCAR  generated by ML method. The ML method could speed up the iteration process of TDEP code. Therefore, I believe that the users of tdeptools expect tdep _ parse _ output.py could read any OUTCAR file, making it more universal.

**Comment by nityasagarjena** [comment]
Hi, you could use the following python parser to get the tdep infiles from VASP MLFF runs,
[tdep_infiles_VASP_MLFF.py.zip](https://github.com/user-attachments/files/15765119/tdep_infiles_VASP_MLFF.py.zip)

**Comment by Guodonglin-cqust** [comment]
yes, it works well, thank you very much @nityasagarjena nityasagarjena

**Comment by phyjz** [comment]
> Hi, you could use the following python parser to get the tdep infiles from VASP MLFF runs, [tdep_infiles_VASP_MLFF.py.zip](https://github.com/user-attachments/files/15765119/tdep_infiles_VASP_MLFF.py.zip)


I discovered that your Python script inaccurately identifies the number of molecular dynamics steps in VASP. I ran 4000 steps with VASP, but it recognized 4296 steps. These additional 296 steps were precisely DFT calculations rather than those fitted using force fields. My VASP version is 6.43.What could be the reason for this?@nityasagarjena
